cerberus/kubernetes/client.py

```diff
@@ -83,29 +84,39 @@ def check_sdn_namespace():
 def monitor_namespace(namespace):
     pods = list_pods(namespace)
     notready_pods = []
+    notready_containers = defaultdict(list)
     for pod in pods:
         try:
             pod_info = cli.read_namespaced_pod_status(pod, namespace,
                                                       pretty=True)
         except ApiException as e:
             logging.error("Exception when calling \
                            CoreV1Api->read_namespaced_pod_status: %s\n" % e)
-        pod_status = pod_info.status.phase
-        if pod_status != "Running" \
-            and pod_status != "Completed" \
-            and pod_status != "Succeeded":
-            notready_pods.append(pod)
-    if len(notready_pods) != 0:
+        pod_status = pod_info.status
+        pod_status_phase = pod_status.phase
+        if pod_status_phase not in ["Completed", "Succeeded"]:
+            for condition in pod_status.conditions:
+                if condition.type == "Ready" and condition.status == "False":
+                    notready_pods.append(pod)
+                if condition.type == "ContainersReady" and condition.status == "False":
+                    for container in pod_status.container_statuses:
+                        if not container.ready:
+                            notready_containers[pod].append(container.name)
+                    if pod_status.init_container_statuses:
+                        for container in pod_status.init_container_statuses:
+                            if not container.ready:
+                                notready_containers[pod].append(container.name)
+    if len(notready_pods) != 0 and len(notready_containers) != 0:
```

Comment:
Maybe change this to "if len(notready_pods) != 0 or len(notready_containers) != 0:"? The pod will be in NotReady state if a container is not ready but better to use or instead of and I guess.

Reply:
Yeah, this makes sense because when the pod is deleted it enters the Pending phase and the containers state isn't known.